proto/api_spec/method.proto

```diff
@@ -102,8 +102,19 @@ message List {
   repeated Data elems = 1;
 }
 
+// Represents a struct or a map.
 message Struct {
+  // Populated if this represents a struct.
   map<string, Data> fields = 1;
+
+  // Populated if this represents a map.
+  MapData map_type = 2;
+}
+
+// Gives the key and value types of a map.
+message MapData {
+  Data key = 1;
```

Comment:
For JSON, the only valid key type is a string, although we could infer a format. Do we want to be more specific here?

Reply:
I think we'll want to use the format of they key (e.g. UUID) to help detect instances of maps.  We might as well keep that data around when we find it.